### ISSUE TITLE
Add a point-based bounty system to Godot, without involving money.

### ISSUE THREAD

**Issue by Sofviic** [godotengine/godot-proposals#2601]
### Describe the project you are working on

N/A

### Describe the problem or limitation you are having in your project

I cannot easily publicize what assets I'd like to see in the future. I wish contributing and helping someone else with making assets for them would be reciprocated by someone.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

A closed economy made-up of fictitious points, that can allow devs to publicly voice their need for a certain yet-to-be asset.
i.e. A bounty system that doesn't involve real money, motivation for completing bounties would only be the ability to put bounties of your own.

This is akin to StackExchange's bounty system, that's based on the concept of "reputation".
see: https://meta.stackexchange.com/questions/16065/how-does-the-bounty-system-work

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

In the following, let **user** means "Godot developers that use the Assets store"
#### In the assets store, 

1. For each user, add a **_points_** fields, that is each user has some amount of _points_ that is kept track of.
2. Add the concept of **_bounties_**, each _bounty_ will have a certain amount of _points_ associated with it, and a list of **criteria** that must be met to claim the _points_.
3. A user may spend some of their _points_ into _bounties_, which would decrease their _user points_, but increase the _bounty points_ (The idea here, is to add _pressure_ that would make it more enticing to complete.)
4. A user may not spend more _points_ than what they have. (That is _point_-debt is a non-option.)
5. A user can only open up a _bounty_ with a minimum threshold of _points_ (ie, a minimum amount of  _points_ required to open up a _bounty_.)

##### Use-Case 1

1. User X wants asset that does A,
2. But no asset exists that does A,
3. User X has 10 _points_,
4. User X puts _bounty_ with 2 _points_,
5. User X now has 8 _points_,
6. User Y has 0 _points_,
7. User Y sees _bounty_ for A,
8. User Y completes asset that does A,
9. User Y claims _bounty_ reward of 2 _points_,
10. User Y has now 2 _points_,
11. User X can now use User Y's public asset to do A.

##### Use-Case 2

1. User Y wants asset that does B,
2. But no asset exists that does B,
3. User Y has 0 _points_,
4. User Y sees _bounty_ for A that pays 2 _points_,
5. User Y completes asset that does A,
6. User Y claims _bounty_ reward of 2 _points_,
7. User Y has now 2 _points_,
8. User Y puts _bounty_ for B with 2 _points_,
9. User Y has 0 _points_,
10. User Z has 6 _points_,
11. User Z sees _bounty_ for B,
12. User Z completes asset that does B,
13. User Z claims _bounty_ reward of 2 _points_,
14. User Z has now 8 _points_,
11. User Y can now use User Z's public asset to do B.

tl;dr People are incentivised to participate, and help others. In return, gaining the ability to incentivise others to do some work for them.

#### Assume this is the question: "If this enhancement will not be used often, can it be worked around with third-party tools?"

Yes possibly, an "asset store"-ish plugin, might be able to implement it.

### Is there a reason why this should be core and not an add-on in the asset library?

It directly address issues on how to increase asset store activity, although as addressed above, it is potentially implementable as a plugin.

footnote: This was directly motivated by discussions in the new `grand-auditorium` in the godot discord server.

**Comment by Calinou** [comment]
In general, [people don't like working for exposure](https://twitter.com/forexposure_txt). While gamification is nice, I wouldn't expect it to make it significantly increase the *quality* of released assets, especially their long-term maintenance. In fact, it could encourage people to create many low-quality assets over fewer well-made ones.

**Comment by viiibe** [comment]
Bad idea imo. It's just incentivizing multiplying the amount of competing work over a single thing, which just makes for unnecessary redundancy.

**Comment by pycbouh** [comment]
This can easily be covered by community efforts, epsecially since there is no money involved. Otherwise, see #2454.

**Comment by dalexeev** [comment]
We just need to add the ability to rate addons (stars like on GitHub or a scale from 1 to 5 like in PlayMarket), without an economic model.

**Comment by Calinou** [comment]
> We just need to add the ability to rate addons (stars like on GitHub or a scale from 1 to 5 like in PlayMarket), without an economic model.

This is already implemented in the new asset libray, but it needs to be deployed first. (I don't have an ETA for this.)

**Comment by Calinou** [comment]
Closing due to lack of support (see reactions on OP).
